snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -248,7 +248,10 @@
         var blockList = getBlockList();
         snippets = snippets.filter(
             function (snippet) {
-                if (blockList.indexOf(snippet.id) === -1) {
+                var snippet_id = parseInt(snippet.id, 10);
+                var campaign = snippet.campaign;
+                var snippetHandle = campaign || snippet_id;
+                if (blockList.indexOf(snippetHandle) === -1) {
```

Comment:
We need to test both against `campaign` and `snippet_id` to be backwards compatible.